lib/zfs.js

```diff
@@ -140,13 +152,41 @@ zpool.status = function (pool, callback) {
  * zpool create command, including log devices, cache devices, and hot spares.
  * The input is an object of the form produced by the disklayout library.
  */
-zpool.create = function (pool, config, callback) {
+zpool.create = function (pool, config, force, encrypt, efi, callback) {
+	var cmd;
 	var args;
 
-	if (arguments.length != 3)
-		throw Error('Invalid arguments, 3 arguments required');
+	if (arguments.length === 3) {
+		callback = force;
+		encrypt = false;
+		force = false;
+		efi = true;
+	} else if (arguments.length === 4) {
+		callback = encrypt;
+		encrypt = false;
+		efi = true;
+	} else if (arguments.length === 5) {
+		efi = true;
+	} else if (arguments.length !== 6) {
+		throw Error('Invalid arguments, 3-6 arguments required');
+	}
 
-	args = [ 'create', pool ];
+	if (encrypt === true) {
+		cmd = exports.paths.kbmadm;
+		args = [ 'create-zpool' ];
+	} else {
+		cmd = exports.paths.zpool;
+		args = [ 'create' ];
+	}
+
+	if (efi === true) {
```

Comment:
Linux check here?!?

Reply:
Yeah.

Reply:
Let the record show these changes will be going into the linuxcn branch.